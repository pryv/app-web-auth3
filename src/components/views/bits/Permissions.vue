<template>
  <v-dialog
    v-model="dialog"
    width="600"
    persistent>
    <v-card>
      <v-card-title
        id="appIdText"
        class="headline grey lighten-2">
        <span><b>{{ accessState.requestingAppId }}</b></span>
      </v-card-title>
      <v-card-text style="text-align:left">
        <h4>Is requesting permission:</h4>
        <ul>
          <li
            v-for="(permission, index) in accessState.requestedPermissions"
            :key="index">
            to {{ permission.level.toLowerCase() }} <u>{{ permission.streamId === '*' ? '* (all data)' : permission.name || permission.defaultName }}</u>
          </li>
        </ul>
        <span
          v-if="expire"
          style="text-align:left">
          <b>will expire after:</b> {{ accessState.expireAfter }}s
        </span>
        <br>
        <span
          v-if="consentMsg"
          v-html="consentMsg"/>
      </v-card-text>
      <v-alert
        v-if="accessMismatch"
        :value="accessMismatch"
        :type="alertType"
        transition="scale-transition"
        style="text-align:left">
        <b>A different access was already given to this app. Do you want to replace it?</b>
      </v-alert>
      <v-divider/>
      <v-card-actions>
        <v-spacer/>
        <v-btn
          id="refusePermissions"
          @click="closeDialog('refused')"
        >Reject</v-btn>
        <v-btn
          id="acceptPermissions"
          @click="closeDialog('accepted')"
        >Accept</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import { marked } from 'marked';
export default {
  props: {
    ctx: { type: Object, default: () => {} },
  },
  data: () => ({
    dialog: true,
    accessState: null,
    checkAppResult: null,
  }),

  computed: {
    consentMsg: function () {
      if (this.ctx.accessState.clientData != null) {
        const description = this.ctx.accessState.clientData['app-web-auth:description'];
        return description != null ? marked.parse(description.content) : '';
      } else {
        return null;
      }
    },
    accessMismatch: function () {
      return (this.ctx.checkAppResult.mismatchingAccess != null);
    },
    expire: function () {
      return (this.ctx.accessState.expireAfter != null);
    },
  },
  created: function () {
    this.accessState = this.ctx.accessState;
    this.checkAppResult = this.ctx.checkAppResult;
  },
  methods: {
    closeDialog (acceptOrReject) {
      this.dialog = false;
      this.$emit(acceptOrReject);
    },
  },
};
</script>
