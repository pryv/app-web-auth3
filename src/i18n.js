import Vue from 'vue';
import VueI18n from 'vue-i18n';

const config = require('../config/i18n.config');

Vue.use(VueI18n);

const i18n = new VueI18n({
  locale: config.locale || 'en',
  fallbackLocale: config.fallbackLocale || 'en',
  messages: loadLocaleMessages(),
});

function loadLocaleMessages () {
  const locales = require.context('../locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};
  locales.keys().forEach(key => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });
  return messages;
};

export default i18n;
