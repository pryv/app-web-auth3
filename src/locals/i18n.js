import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

export const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages: {
    en: {
      // Authorization
      signin: 'Sign in',
      accessRequest: 'App <b>{appId}</b> is requesting :',
      newPermission: 'A permission on stream <b>{stream}</b> with level <b>{level}</b>.',
      acceptAccess: 'Accept',
      rejectAccess: 'Reject',

      // Register User
      register: 'Register a new user',
      chooseHosting: 'Choose hosting',
      hosting: 'Hosting',
      createUser: 'Create user',
      userCreated: 'New user successfully created: ',

      // Reset Password
      forgotPass: 'Forgot password',
      resetPass: 'Reset password',
      requestResetPass: 'Reset',
      setNewPass: 'Set a new password',
      changePass: 'Change',
      resetInstructions: 'We have sent password reset instructions to your e-mail address.',
      passwordChanged: 'Your password have been successfully changed.',

      // Signin Hub
      newToPryv: 'New to Pryv ? Create an account',

      // Page not found
      notFound: '404 - Page not found',
      goToPryv: 'Go to my Pryv',
      signinHub: 'Signin hub',

      // Commons
      requireField: 'This field is required',
      username: 'Username',
      usernameOrEmail: 'Username or email',
      email: 'E-mail',
      invalidEmail: 'E-mail address is invalid',
      pass: 'Password',
      confirmPass: 'Password confirmation',
      requirePass: 'Password is required',
      mismatchPass: 'Password confirmation does not match',
      cancel: 'Cancel',
      reachUs1: 'Feel free to reach our',
      reachUs2: 'if you have questions.',
      clear: 'Clear form',
      terms1: 'By registering you agree with our',
      terms2: 'terms of use',
      backToSignin: 'Go back to Sign in',
    },
    fr: {
      // Authorization
      signin: 'Connection',
      accessRequest: 'L\' app <b>{appId}</b> demande :',
      newPermission: 'Une permission sur le stream <b>{stream}</b> avec le niveau d\'accès <b>{level}</b>.',
      acceptAccess: 'Accepter',
      rejectAccess: 'Refuser',

      // Register User
      register: 'Créer un nouvel utilisateur',
      chooseHosting: 'Choisir un hébergement',
      hosting: 'Hébergement',
      createUser: 'Créer',
      userCreated: 'Nouvel utilisateur créé avec succès :',

      // Reset Password
      forgotPass: 'Mot de passe oublié',
      resetPass: 'Réinitialiser le mot de passe',
      requestResetPass: 'Réinitialiser',
      setNewPass: 'Définir un nouveau mot de passe',
      changePass: 'Changer',
      resetInstructions: 'Nous avons envoyé les instructions pour la réinitialisation du mot de passe à votre adresse email.',
      passwordChanged: 'Votre mot de passe à été changé avec succès.',

      // Signin Hub
      newToPryv: 'Nouveau sur Pryv? Créez un nouveau compte',
      goToPryv: 'Aller à mon compte Pryv',
      signinHub: 'Hub de connection',

      // Page not found
      notFound: '404 - Page indisponible',

      // Commons
      requireField: 'Ce champ est requis',
      username: 'Nom d\'utilisateur',
      usernameOrEmail: 'Nom d\'utilisateur ou adresse email',
      email: 'Adresse email',
      invalidEmail: 'L\'adresse email est invalide',
      pass: 'Mot de passe',
      confirmPass: 'Confirmation du mot de passe',
      requirePass: 'Le mot de passe est requis',
      mismatchPass: 'La confirmation du mot de passe ne correspond pas',
      cancel: 'Annuler',
      reachUs1: 'N\'hésitez pas à contacter notre',
      reachUs2: 'si vous avez des questions.',
      clear: 'Effacer',
      terms1: 'En vous enregistrant, vous acceptez nos',
      terms2: 'termes et conditions.',
      backToSignin: 'Revenir à la page de connection',
    },
  },
});
