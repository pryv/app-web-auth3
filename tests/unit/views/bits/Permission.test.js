import { shallowMount } from '@vue/test-utils';
import Permissions from '@/components/views/bits/Permissions';
import Vue from 'vue';
import Vuetify from 'vuetify';
import {i18n} from '@/locals/i18n.js';

Vue.use(Vuetify);

describe('Permissions.test.js', () => {
  let wrapper;
  const appId = 'testApp';
  const permissions = [
    {name: 'Diary', level: 'read'},
    {defaultName: 'Work', level: 'manage'},
  ];

  beforeAll(() => {
    wrapper = shallowMount(Permissions, {
      propsData: {
        permissionsList: permissions,
        appId: appId,
        accept: () => {},
        refuse: () => {},
      },
      mocks: {
        $t: (...args) => i18n.t(...args),
      },
    });
  });

  it('renders correctly (snapshots matching)', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('shows the id of the requesting app', () => {
    const appIdText = wrapper.find('#appIdText');
    expect(appIdText.html()).toContain(`App <b>${appId}</b> is requesting :`);
  });

  it('shows a list of all requested permissions', () => {
    const list = wrapper.find('ul').html();
    expect(list).toContain(`A permission on stream <b>Diary</b> with level <b>READ</b>.`);
    expect(list).toContain(`A permission on stream <b>Work</b> with level <b>MANAGE</b>.`);
  });
});
