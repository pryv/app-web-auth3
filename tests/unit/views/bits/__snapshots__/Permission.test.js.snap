// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permissions.test.js renders correctly (snapshots matching) 1`] = `
<v-dialog-stub
  accept="function accept() {}"
  contentclass=""
  maxwidth="none"
  origin="center center"
  persistent="true"
  refuse="function refuse() {}"
  transition="dialog-transition"
  value="true"
  width="600"
>
  <v-card-stub
    tag="div"
  >
    <v-card-title-stub
      class="headline grey lighten-2"
      id="appIdText"
    >
      <span>
        App 
        <b>
          testApp
        </b>
         is requesting : 
      </span>
    </v-card-title-stub>
     
    <v-card-text-stub>
      <ul>
        <li>
          
          A permission on stream 
          <b>
            Diary
          </b>
           with level 
          <b>
            READ
          </b>
        </li>
        <li>
          
          A permission on stream 
          <b>
            Work
          </b>
           with level 
          <b>
            MANAGE
          </b>
        </li>
      </ul>
    </v-card-text-stub>
     
    <v-card-text-stub />
     
    <v-divider-stub />
     
    <v-card-actions-stub>
      <v-spacer-stub />
       
      <v-btn-stub
        activeclass="v-btn--active"
        id="refusePermissions"
        tag="button"
        type="button"
      >
        Reject
      </v-btn-stub>
       
      <v-btn-stub
        activeclass="v-btn--active"
        id="acceptPermissions"
        tag="button"
        type="button"
      >
        Accept
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
